src/app/modules/explorer/transactions/explorerTransactions.controller.js

```diff
@@ -35,29 +35,34 @@ class ExplorerTransactionsCtrl {
 
         // Get transactions
         this.getTransactions(false);
+
+        // Watch node change
+        $scope.$watch(() => this._Wallet.node, (val) => {
+            if (!val) return;
+            this.transactions = [];
+            this.noMoreTxes = false;
+            this.currentPage = 0;
+            this.getTransactions(false);
+        }, true);
     }
 
     //// Module methods region ////
 
     /**
      * Get transactions of the account
      */
-    getTransactions(isUpdate, txHash) {
-        let endpoint;
-        if (this._Wallet.network === nem.model.network.data.mainnet.id) {
-            endpoint = nem.model.objects.create("endpoint")("http://hugealice.nem.ninja", 7890);
-        } else {
-            endpoint = this._Wallet.node;
-        }
-       let obj = {
+    getTransactions(isUpdate, txId) {
+        let obj = {
             'params': {
                 'address': this._Wallet.currentAccount.address,
-                'hash': txHash ? txHash : '',
                 'pageSize': isUpdate ? 100 : 50
             }
         };
-        return this._$http.get(endpoint.host + ':' + endpoint.port + '/account/transfers/all', obj).then((res) => {
-            if(isUpdate) {
+        if (isUpdate) {
+            obj['params']['id'] = txId;
+        }
+        return this._$http.get(this._Wallet.node.host + ':' + this._Wallet.node.port + '/account/transfers/all', obj).then((res) => {
```

Comment:
request is now made to the node that is used in the wallet (better than hardcoded alice)